packages/jellyfish-api-core/__tests__/category/icxorderbook/submitDFCHTLC.test.ts

```diff
@@ -0,0 +1,338 @@
+
+  it('should submit DFC HTLC with input utxos for a DFC buy offer', async () => {
+    const { order, createOrderTxId } = await icxSetup.createDFISellOrder('BTC', accountDFI, '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941', new BigNumber(15), new BigNumber(0.01))
+    const { makeOfferTxId } = await icxSetup.createDFIBuyOffer(createOrderTxId, new BigNumber(0.10), accountBTC)
+
+    const accountDFIBeforeDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // create DFCHTLC - maker
+    const DFCHTLC: HTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(10), // in  DFC
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 500
+    }
+
+    // input utxos
+    const utxos = await container.call('listunspent', [1, 9999999, [accountDFI], true])
+    const inputUTXOs: UTXO[] = utxos.map((utxo: UTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+    const DFCHTLCTxId = (await client.icxorderbook.submitDFCHTLC(DFCHTLC, inputUTXOs)).txid
+    await container.generate(1)
+
+    const accountDFIAfterDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // maker fee should be reduced from accountDFIBeforeDFCHTLC
+    expect(accountDFIAfterDFCHTLC[idDFI]).toStrictEqual(accountDFIBeforeDFCHTLC[idDFI].minus(0.01))
+
+    // List htlc
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.call('icx_listhtlcs', [listHTLCOptions], 'bignumber')
+    expect(Object.keys(HTLCs).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect(HTLCs[DFCHTLCTxId] as ICXDFCHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.DFC,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: DFCHTLC.amount,
+        amountInEXTAsset: DFCHTLC.amount.multipliedBy(order.orderPrice),
+        hash: DFCHTLC.hash,
+        timeout: new BigNumber(DFCHTLC.timeout as number),
+        height: expect.any(BigNumber),
+        refundHeight: expect.any(BigNumber)
+      }
+    )
+  })
+
+  it('should return an error when submiting DFC HTLC with invalid HTLC.offerTx', async () => {
+    const { createOrderTxId } = await icxSetup.createDFISellOrder('BTC', accountDFI, '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941', new BigNumber(15), new BigNumber(0.01))
+    const { makeOfferTxId } = await icxSetup.createDFIBuyOffer(createOrderTxId, new BigNumber(0.10), accountBTC)
+
+    const accountDFIBeforeDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // create DFCHTLC with invalid offerTx "INVALID_OFFER_TX_ID" - maker
+    const DFCHTLC: HTLC = {
+      offerTx: 'INVALID_OFFER_TX_ID',
```

Comment:
try `1111...` see what the error says

Reply:
maybe should have another test for it, one being an invalid hex, another being txid not found

Reply:
added test for `123`
error message
```
RpcApiError: 'offerTx (0000000000000000000000000000000000000000000000000000000000000123) does not exist', code: -8, method: icx_submitdfchtlc
```
The reason for this behaviour can be clearly seen from C++ code. 
I've informed about this to Mihailo in `listorders`